File: 134-gas-station/gas-station.java
class Solution {
    public int canCompleteCircuit(int[] gas, int[] cost) {
        int[] arr = new int[gas.length];
        int n = gas.length;
        int gasT = 0;
        int costT = 0;
        for (int i = 0; i < n; i++) {
            arr[i] = gas[i] - cost[i];
            gasT += gas[i];
            costT += cost[i];
        }
        if (costT > gasT) {
            return -1;
        }
        int sum = 0;
        int start = 0;
        for (int i = 0; i < n; i++) {
            if (sum == 0) {
                start = i;
            }
            sum += arr[i];
            if (sum < 0) {
                sum = 0;
            }
        }
        return start;
    }
}

File: 134-gas-station/gas-station.py
class Solution:
    def canCompleteCircuit(self, gas: List[int], cost: List[int]) -> int:
        gasCost = []
        for i in range(len(gas)):
            gasCost.append(gas[i] - cost[i])
        
        if sum(gasCost) < 0:
            return -1
        print(gasCost)
        currSum = 0
        maxGas = float('-inf')
        ans = 0
        for i in range(len(gasCost)):
            if currSum == 0:
                ans = i
            currSum += gasCost[i]
            maxGas = max(maxGas, currSum)
            if currSum < 0:
                currSum = 0 
        print(maxGas)
        return ans